Service/ContentTypeService.php

```diff
@@ -426,10 +430,30 @@ public function getAllTypes()
         return implode(',', array_keys($this->contentTypeArrayByName));
     }
 
-    public function contentTypeFromJson(string $json, Environment $environment): ContentType
+
+    public function updateFromJson(ContentType $contentType, string $json, bool $isDeleteExitingTemplates, bool $isDeleteExitingViews): void
+    {
+        $this->deleteFields($contentType);
+        if ($isDeleteExitingTemplates) {
+            $this->deleteTemplates($contentType);
+        }
+        if ($isDeleteExitingViews) {
+            $this->deleteViews($contentType);
+        }
+
+        $environment = $contentType->getEnvironment();
+        if (! $environment instanceof Environment) {
+            throw new NotFoundHttpException('Environment not found');
+        }
+
+        $updatedContentType = $this->contentTypeFromJson($json, $environment, $contentType);
+        $this->importContentType($updatedContentType);
+    }
+
+    public function contentTypeFromJson(string $json, Environment $environment, ContentType $contentType = null): ContentType
     {
         $meta = JsonClass::fromJsonString($json);
-        $contentType = $meta->jsonDeserialize();
+        $contentType = $meta->jsonDeserialize($contentType);
```

Comment:
As discussed in the JsonClass file, this should be a call to $meta->deserializeProperties($contentType). However this should not be called from here (as this function should only care to create contentTypeFromJson (factor out the code you need to a separate function?)